dsc_lib/src/configure/contraints.rs

```diff
@@ -0,0 +1,170 @@
+    if let Some(min_length) = constraint.min_length {
+        if value.is_string() {
+            let Some(value) = value.as_str() else {
+                return Err(DscError::Validation(format!("Parameter '{name}' has minimum length constraint but is null")));
+            };
+
+            if value.len() < usize::try_from(min_length)? {
+                return Err(DscError::Validation(format!("Parameter '{name}' has minimum length constraint of {min_length} but is {0}", value.len())));
+            }
+        }
+        else if value.is_array() {
+            let Some(value) = value.as_array() else {
+                return Err(DscError::Validation(format!("Parameter '{name}' has minimum length constraint but is null")));
+            };
+
+            if value.len() < usize::try_from(min_length)? {
+                return Err(DscError::Validation(format!("Parameter '{name}' has minimum length constraint of {min_length} but is {0}", value.len())));
+            }
+        }
+        else {
+            return Err(DscError::Validation(format!("Parameter '{name}' has minimum length constraint but is not a string or array")));
+        }
+    }
+
+    if let Some(max_length) = constraint.max_length {
+        if value.is_string() {
+            let Some(value) = value.as_str() else {
+                return Err(DscError::Validation(format!("Parameter '{name}' has maximum length constraint but is null")));
+            };
+
+            if value.len() > usize::try_from(max_length)? {
+                return Err(DscError::Validation(format!("Parameter '{name}' has maximum length constraint of {max_length} but is {0}", value.len())));
+            }
+        }
+        else if value.is_array() {
+            let Some(value) = value.as_array() else {
+                return Err(DscError::Validation(format!("Parameter '{name}' has maximum length constraint but is null")));
+            };
+
+            if value.len() > usize::try_from(max_length)? {
+                return Err(DscError::Validation(format!("Parameter '{name}' has maximum length constraint of {max_length} but is {0}", value.len())));
+            }
+        }
+        else {
+            return Err(DscError::Validation(format!("Parameter '{name}' has maximum length constraint but is not a string or array")));
+        }
+    }
+
+    Ok(())
+}
+
+/// Checks that the given value matches the given number value constraints.
+///
+/// # Arguments
+///
+/// * `name` - The name of the parameter.
+/// * `value` - The value of the parameter.
+/// * `constraint` - The constraints on the parameter.
+///
+/// # Returns
+///
+/// * `Ok(())` if the value matches the constraints.
+/// * `Err(DscError::Validation)` if the value does not match the constraints.
+///
+/// # Errors
+///
+/// * `DscError::Validation` if the value does not match the constraints.
+pub fn check_number(name: &str, value: &Value, constraint: &Parameter) -> Result<(), DscError> {
```

Comment:
I would rename this to be more specific like something like "check_number_limits" or "check_max_min" .

Reply:
agree on naming, will make this change.